message/validation/consensus_validation.go

```diff
@@ -244,7 +244,7 @@ func (mv *messageValidator) validateSignerBehaviorConsensus(
 	}
 
 	if msgSlot == signerState.Slot && msgRound == signerState.Round {
-		if mv.hasFullData(signedMsg) && signerState.ProposalData != nil && !bytes.Equal(signerState.ProposalData, signedMsg.FullData) {
+		if mv.needDuplicatedFullDataCheck(signedMsg) && signerState.ProposalData != nil && !bytes.Equal(signerState.ProposalData, signedMsg.FullData) {
```

Comment:
so before the error occured because the fullData of a different message type (non-proposal)?

Reply:
The error occurred before because `fullData` was checked for:
- proposal
- decided
- round change

Here error doesn't occur and `fullData` is checked for:
- proposal
- decided

So `fullData` duplication check for round change caused issues.

Reply:
I think the logic is fine.
I am trying to understand why the error showed up though...
You usually saw the error on the first round correct (that's what I remember but not sure)?

It makes sense that maybe on a new round we get a different full data...
But not sure why it happened on a first round

Reply:
I usually saw error happening on proposal or decided messages in round 2 because full data was compared to round change messages for round 2

Reply:
this makes more sense I think

Reply:
I remember message type used to say commit on most of this error, so it happened when we had round change after a commit?

Reply:
no, it happened on commit after round change, because round change full data was remembered, so commit full data was compared to round change full data